### PR TITLE
Add new actions Submit File, Submit URL and Get Sandbox Report

### PR DESCRIPTION
**Pull request by gusingh-r7**
### Description

Describe the proposed changes:

  - Add new actions Submit File, Submit URL and Get Sandbox Report

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [x] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by gusingh-r7** [comment]
```
INFO[0000] Running command:  docker run --rm -i rapid/anomali_threatstream:3.1.0 run < tests/submit_url.json
INFO[0002]  Output:
{
  "reports": [
    {
      "detail": "/api/v1/submit/178/report/",
      "id": 178,
      "platform": "WINDOWS7",
      "status": "/api/v1/submit/178/"
    }
  ],
  "success": true
}
~/go/src/github.com/rapid7/insightconnect-plugins/anomali_threatstream$ icp run -R tests/submit_file.json -j
INFO[0000] Running command:  docker run --rm -i rapid/anomali_threatstream:3.1.0 run < tests/submit_file.json
INFO[0003]  Output:
{
  "reports": [
    {
      "detail": "/api/v1/submit/179/report/",
      "id": 179,
      "platform": "WINDOWS7",
      "status": "/api/v1/submit/179/"
    }
  ],
  "success": true
}
INFO[0000] Running command:  docker run --rm -i rapid7/anomali_threatstream:3.1.0 run < tests/get_sandbox_report.json
{
  "sandbox_report": {
    "domains": [
      "star-mini.c10r.facebook.com",
      "pagead46.l.doubleclick.net",
      "star.c10r.facebook.com",
      "us-u.openx.net"
    ],
    "info": {
      "category": "URL",
      "confidence": 0,
      "duration": 302,
      "ended": "2019-12-04 21:09:32",
      "is_malicious": false,
      "is_suspicious": false,
      "is_unknown": false,
      "started": "2019-12-04 21:04:30"
    },
    "screenshots": [
      "http://svlpartner-optic.s3.amazonaws.com/userUploads/2019-12-04/20191204_201304_userId-121_tmpsandbox-report-full-PaRfUDscreenshot_00.png",
      "http://svlpartner-optic.s3.amazonaws.com/userUploads/2019-12-04/20191204_201305_userId-121_tmpsandbox-report-full-PaRfUDscreenshot_01.png",
      "http://svlpartner-optic.s3.amazonaws.com/userUploads/2019-12-04/20191204_201305_userId-121_tmpsandbox-report-full-PaRfUDscreenshot_02.png"
    ],
    "signatures": [
      {
        "data": [
          {
            "operation": "Window detected",
            "process": "Window Recorder",
            "value": "More than 3 window changes detected"
          }
        ],
        "description": "Found graphical window changes (likely an installer)",
        "severity": "-2.0"
      },
      {
        "data": [
          {
            "operation": "File opened",
            "process": "C:\\Program Files (x86)\\Internet Explorer\\iexplore.exe",
            "value": "C:\\Program Files (x86)\\Java\\jre1.8.0_191\\bin\\msvcr100.dll"
          }
        ],
        "description": "Uses new MSVCR Dlls",
        "severity": "-1.0"
      }
    ]
}
```

File: anomali_threatstream/plugin.spec.yaml [hunk 1]
```diff
@@ -324,3 +404,134 @@ actions:
...
+        type: boolean
+        required: false
+      reports:
+        title: Reports
+        description: Reports containing submission details
+        type: "[]report"
+        required: false
+  get_sandbox_report:
+    title: Get Sandbox Report
+    description: Get sandbox report
```
Comment (lmilby-r7) [inline anomali_threatstream/plugin.spec.yaml]:
```suggestion
    description: Get a sandbox report
```

File: anomali_threatstream/plugin.spec.yaml [hunk 2]
```diff
@@ -324,3 +404,134 @@ actions:
...
+        type: boolean
+        required: false
+      reports:
+        title: Reports
+        description: Reports containing submission details
+        type: "[]report"
+        required: false
+  submit_url:
+    title: Submit URL
+    description: Submit a URL to ThreatStream sandbox
```
Comment (lmilby-r7) [inline anomali_threatstream/plugin.spec.yaml]:
```suggestion
    description: Submit a URL to a ThreatStream sandbox
```

File: anomali_threatstream/plugin.spec.yaml [hunk 3]
```diff
@@ -324,3 +404,134 @@ actions:
         description: Results returned
         type: '[]result'
         required: false
+  submit_file:
+    title: Submit File
+    description: Submit a file to ThreatStream sandbox
```
Comment (lmilby-r7) [inline anomali_threatstream/plugin.spec.yaml]:
```suggestion
    description: Submit a file to a ThreatStream sandbox
```

File: anomali_threatstream/komand_anomali_threatstream/actions/submit_file/action.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='submit_file',
+                description=Component.DESCRIPTION,
+                input=SubmitFileInput(),
+                output=SubmitFileOutput())
+
+    def run(self, params={}):
+        self.request = copy(self.connection.request)
+        self.request.url, self.request.method = self.request.url + "/submit/new/", "POST"
```
Comment (lmilby-r7) [inline anomali_threatstream/komand_anomali_threatstream/actions/submit_file/action.py]:
```suggestion
        self.request.url, self.request.method = f"{self.request.url}/submit/new/", "POST"
```

File: anomali_threatstream/komand_anomali_threatstream/actions/get_sandbox_report/action.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        super(self.__class__, self).__init__(
+                name='get_sandbox_report',
+                description=Component.DESCRIPTION,
+                input=GetSandboxReportInput(),
+                output=GetSandboxReportOutput())
+
+    def run(self, params={}):
+        self.request = copy(self.connection.request)
+        report_id = params.get(Input.REPORT_ID)
+        self.request.url, self.request.method = self.request.url + f"/submit/{report_id}/report/", "GET"
```
Comment (lmilby-r7) [inline anomali_threatstream/komand_anomali_threatstream/actions/get_sandbox_report/action.py]:
```suggestion
        self.request.url, self.request.method = f"{self.request.url}/submit/{report_id}/report/", "GET"
```

File: anomali_threatstream/plugin.spec.yaml [hunk 6]
```diff
@@ -221,6 +221,86 @@ types:
       title: Import Session ID
       description: ID for import session
       type: string
+  report:
+    status:
+      title: Status
+      decription: Link to submission status
+      type: string
+    detail:
+      title: Detail
```
Comment (mrinehart-r7) [inline anomali_threatstream/plugin.spec.yaml]:
```suggestion
      title: Details
```

File: anomali_threatstream/plugin.spec.yaml [hunk 7]
```diff
@@ -324,3 +404,134 @@ actions:
...
+    input:
+      report_id:
+        title: Report ID
+        description: Report ID
+        type: string
+        required: true
+    output:
+      sandbox_report:
+        title: Sandbox Report
+        description: Sandbox Report
```
Comment (mrinehart-r7) [inline anomali_threatstream/plugin.spec.yaml]:
```suggestion
        description: Sandbox report
```

**Comment by lmilby-r7** [comment]
Grep for Anamoli `grep -rnw . -e 'Anamoli' ` and change all to `Anomali `

**Pull request merged**